src/torrent.ts

```diff
@@ -64,35 +66,43 @@ export async function snatch(
 		});
 	} catch (e) {
 		if (e.name === "AbortError") {
-			logger.error(`snatch timed out from ${tracker}: ${url}`);
+			logger.error(
+				`Snatch timed out from ${tracker} for ${candidate.name}`,
+			);
+			logger.debug(`${candidate.name}: ${url}`);
 			return resultOfErr(SnatchError.ABORTED);
 		} else if (isMagnetRedirectError(e)) {
 			logger.error(
-				`Unsupported: magnet link detected at ${tracker}: ${url}`,
+				`Unsupported: magnet link detected from ${tracker} for ${candidate.name}`,
 			);
+			logger.debug(`${candidate.name}: ${url}`);
 			return resultOfErr(SnatchError.MAGNET_LINK);
 		}
-		logger.error(`failed to access ${tracker}: ${url}`);
-		logger.debug(e);
+		logger.error(`Failed to access ${tracker} for ${candidate.name}`);
+		logger.debug(`${candidate.name}: ${url} - ${e}`);
```

Comment:
this means we lose the `e` stack trace. can we split this into two lines?
```suggestion
		logger.debug(`${candidate.name}: ${url}`);
		logger.debug(e);
```